Assets/Scripts/Gameplay/GameplayObjects/NetworkPickUpState.cs

```diff
@@ -0,0 +1,10 @@
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    public class NetworkPickUpState : NetworkBehaviour
+    {
+        public NetworkVariable<ulong> heldObject;
```

Comment:
NetworkObjectReference could be used instead of ulong

Reply:
I should note that I've went back and reverted it back to a `NetworkVariable<ulong>` since `NetworkObjectReference` does not have a OnValueChanged callback.